### DISCUSSION TITLE
Is there any way to add more parameters to the profile object?

### DISCUSSION THREAD

**Discussion by sawa-ko** [nextauthjs/next-auth#5642] [Help] [answered]
### Question 💬

Currently when I use the profile method of my custom provider, only the properties id, email, image and name are obtained in the hook useSession, if I pass any other parameter than any of them, the parameter does not exist in the hook but the other parameters do exist.

I don't know if this works that way, but if so, this augmentation makes no sense https://next-auth.js.org/getting-started/typescript#main-module.

### How to reproduce ☕️

```
	profile(profile) {
		return {
			id: profile.id,
			name: profile.username,
			image: profile.image_url,
			custom_property: "hello" // this is undefined in useSession hook
		};
	}
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by ThangHuuVu** [comment, marked as answer]
If you want to add properties to the `useSession` hook's `data`, normally you'd only need to use the `callbacks.session` option:
```
callbacks: {
    session(params) {
      params.session.foo = "bar"
      return params.session
    },
  },
```
Let me know if this helps! 🙌
